virttest/utils_test/libvirt.py

```diff
@@ -1438,47 +1438,55 @@ def create_hostdev_xml(pci_id, boot_order=None, xmlfile=True,
     return hostdev_xml.xml
 
 
-def create_controller_xml(contr_dict, oper="get_xml", vm_name=None):
+def add_controller(vm_name, contr):
     """
-    Create controller xml or add controller
+    Add a specified controller to the vm xml
+    :param vm_name: The vm name to be added with a controller
+    :param contr: The controller object to be added
+    :return: The newly added controller object
+    """
+    vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    org_controllers = vmxml.get_controllers(contr.type)
+    vmxml.add_device(contr)
+    vmxml.sync()
+    vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    updated_controllers = vmxml.get_controllers(contr.type)
+    for contr in updated_controllers:
```

Comment:
The block local variable has the same name with parameter contr. 
I know it prioritize over parameter variable,but it is better not override it in order to avoid confusion

Reply:
Fix the original codes. Update.